benchmarks/MOABB/dataio/datasets.py

```diff
@@ -0,0 +1,468 @@
+        cache_path = mne_path / cache_dir
+
+        subject_list = (
+            subjects if subjects is not None else dataset.subject_list
+        )
+        dataset.download(subject_list)
+
+        # Convert from MOABB format to BIDS
+        for sub in subject_list:
+            dataset.get_data(
+                subjects=[sub],
+                cache_config=dict(use=True, save_raw=True, path=mne_path),
+            )
+
+        @provides("dataset")
+        def _add_dataset():
+            return dataset
+
+        # Dynamically add the dataset reference
+        dynamic_items = list(cls_kwargs.pop("dynamic_items", []))
+        dynamic_items.insert(0, _add_dataset)
+
+        return cls.from_bids(
+            bids_path=cache_path,
+            json_path=json_path,
+            subjects=subjects,
+            dynamic_items=dynamic_items,
+            **cls_kwargs,
+        )
+
+    def __getitem__(self, index) -> RawEEGSample:
+        return super().__getitem__(index)  # type: ignore
+
+    @classmethod
+    def load_or_create_json_data_from_bids(
+        cls,
+        bids_path: BIDSPath,
+        json_path: Path | str,
+        subjects=None,
+    ) -> dict[str, dict]:
+        """Indexes the BIDS directory and saves the result to a JSON file, or loads
+        the index from JSON if it already exists.
+
+        Arguments
+        ---------
+        bids_path : BIDSPath
+            The BIDS root directory to index
+        json_path : Path, str
+            The .json path where the index will be cached
+        subjects : list[str], optional
+            Index only a subset of subjects. Defaults to all subjects.
+
+        Returns
+        -------
+        dict[str, dict]
+            Returns a mapping from unique sample ID to sample metadata
+        """
+        json_path = Path(json_path)
+        if json_path.exists():
+            with json_path.open() as fp:
+                json_data = json.load(fp)
+        else:
+            json_data = cls.json_data_from_bids_path(bids_path)
+
+            with json_path.open("w") as fp:
+                json.dump(json_data, fp)
+
+        if subjects is not None:
+            json_data = {
+                uid: data
+                for uid, data in json_data.items()
+                if data.get("subject") in subjects
+            }
+
+        return json_data
+
+    @classmethod
+    def json_data_from_bids_path(cls, bids_path: BIDSPath) -> dict[str, dict]:
+        """Indexes all BIDSPaths which match the desired suffix, and returns as a dict.
+
+        Arguments
+        ---------
+        bids_path : BIDSPath
+            The BIDS root directory to index.
+
+        Returns
+        -------
+        dict[str, dict]
+            Returns a mapping from unique sample ID to sample metadata
+        """
+        json_data: dict[str, dict] = {}
+
+        matched_paths: Iterable[BIDSPath] = bids_path.update(
+            suffix=cls.suffix
+        ).match(ignore_json=True)
```

Comment:
```suggestion
        matched_paths: Iterable[BIDSPath] = mne_bids.find_matching_paths(
            root=bids_path.root,
        )
```
this will return everything